node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -335,32 +331,71 @@ object WebApi {
   private def exprFromBundleProto(b: Bundle): Option[RhoExpr] = exprFromParProto(b.body)
 
   private def mkUnforgExpr(f: String => RhoUnforg, bs: ByteString): ExprUnforg =
-    ExprUnforg(f(toHex(bs)))
+    ExprUnforg(f(bs.toHexString))
 
   // RhoExpr to protobuf
 
-  private def unforgToUnforgProto(unforg: RhoUnforg): GUnforgeable.UnfInstance = unforg match {
-    case UnforgPrivate(name)  => GPrivateBody(GPrivate(name.unsafeHexToByteString))
-    case UnforgDeploy(name)   => GDeployIdBody(GDeployId(name.unsafeHexToByteString))
-    case UnforgDeployer(name) => GDeployerIdBody(GDeployerId(name.unsafeHexToByteString))
+  /**
+    * Private definition of [[Monoid]] for [[Par]] with the implementation of
+    * combining fields used by WebApi representation of Par [[RhoExpr]].
+    */
+  implicit private object ParMonoid extends Monoid[Par] {
+    override def empty: Par = VectorPar()
+
+    override def combine(x: Par, y: Par): Par =
+      x.copy(
+        x.sends ++ y.sends,
+        x.receives ++ y.receives,
+        x.news ++ y.news,
+        x.exprs ++ y.exprs,
+        x.matches ++ y.matches,
+        x.unforgeables ++ y.unforgeables,
+        x.bundles ++ y.bundles,
+        x.connectives ++ y.connectives
+      )
+  }
+
+  /**
+    * Converts [[RhoExpr]] WebApi representation to protobuf [[Par]] type.
+    */
+  def rhoExprToParProto(exp: RhoExpr): Par = exp match {
```

Comment:
Generative tests can be written as a round trip from `Par => Par` composition of `exprFromParProto andThen rhoExprToParProto`.